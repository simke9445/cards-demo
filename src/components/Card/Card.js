import React from 'react';
import { FormattedMessage } from "react-intl";
import PropTypes from 'prop-types';

import { Icon, Tooltip } from "@blueprintjs/core";

const styles = {
    container: {
      flex: '1',
      width: '100%',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      flexDirection: 'column',
      height: '100%',
    },
    card: {
      header: {
        backgroundColor: '#A06FEF',
      },
      container: {
        width: '340px',
        height: '500px',
        padding: '0',
        display: 'flex',
        flexDirection: 'column',     
        wordWrap: 'break-word',
        boxShadow: 'rgba(0, 0, 0, 0.14) 0px 0em 2em 2px' +
                  ', rgb(242, 242, 242) 0px 0.75em 0px -0.25em' +
                  ', rgba(0, 0, 0, 0.09) 0px 1.5em 0.1875em -0.5em' +
                  ', rgb(229, 229, 229) 0px 1em 0px -0.5em' +
                  ', rgba(0, 0, 0, 0.1) 0px 1em 0.1875em -0.5em',
      },
      content: {
        padding: '24px',
        flex: 1,
        display: 'flex',
        justifyContent: 'center',
        flexDirection: 'column',
      },
      next: {
        container: {
          height: '30%',
          display: 'flex',
        },
        icon: {
          borderRadius: '50%',
          backgroundColor: '#DDDDDD',
          width: '20%',
          height: '58%',
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center',
          margin: 'auto',
          color: 'white',
          cursor: 'pointer',
        },
      },
    },
    footer: {
      height: '10%',
      justifyContent: 'center',
      alignItems: 'center',
      display: 'flex',
      backgroundColor: 'white',
      borderTop: '1px solid #DCDCDC',
    },
    tooltip: {
      color: '#AFB0B2',
    }
  };

const Card = ({ children, onNext, hasNavigation, hasTooltip, tooltipMessage }) =>     
  <div style={styles.container}>
    <div style={styles.card.container} className="pt-card pt-elevation-2">
      <div className="pt-callout" style={styles.card.header}>
        <h5 style={{ color: 'white' }}><FormattedMessage id="card.header"/></h5>
      </div>
      <div style={styles.card.content}>
        {children}
        {hasNavigation &&
          <div style={styles.card.next.container}>
            <Icon onClick={onNext} iconName="arrow-right" style={styles.card.next.icon}/>
          </div>}
      </div>
      <div className="pt-callout" style={styles.footer}>
        {hasTooltip && 
          <Tooltip
              className="pt-tooltip-indicator"
              content={
                <em>
                  {tooltipMessage}
                </em>
              }
          >
            <span style={styles.tooltip}><FormattedMessage id="card.footer"/></span>
          </Tooltip>}
      </div>
    </div>
  </div>


Card.propTypes = {
  onNext: PropTypes.func,
  hasNavigation: PropTypes.bool,
  hasTooltip: PropTypes.bool,
  tooltipMessage: PropTypes.string,
};

export default Card;