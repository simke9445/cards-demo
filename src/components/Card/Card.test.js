import React from 'react';
import { Card } from './Card';
import { shallow } from 'enzyme';

it('Card renders without crashing', () => {
  const wrapper = shallow(
    <Card hasNavigation={true} onNext={() => null}>
      <div>test</div>
    </Card>
  );

  expect(wrapper).toMatchSnapshot();
});
