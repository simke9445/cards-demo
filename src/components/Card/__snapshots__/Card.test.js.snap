// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders without crashing 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flex": "1",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <div
    className="pt-card pt-elevation-2"
    style={
      Object {
        "boxShadow": "rgba(0, 0, 0, 0.14) 0px 0em 2em 2px, rgb(242, 242, 242) 0px 0.75em 0px -0.25em, rgba(0, 0, 0, 0.09) 0px 1.5em 0.1875em -0.5em, rgb(229, 229, 229) 0px 1em 0px -0.5em, rgba(0, 0, 0, 0.1) 0px 1em 0.1875em -0.5em",
        "display": "flex",
        "flexDirection": "column",
        "height": "500px",
        "padding": "0",
        "width": "340px",
        "wordWrap": "break-word",
      }
    }
  >
    <div
      className="pt-callout"
      style={
        Object {
          "backgroundColor": "#A06FEF",
        }
      }
    >
      <h5
        style={
          Object {
            "color": "white",
          }
        }
      >
        <FormattedMessage
          id="card.header"
          values={Object {}}
        />
      </h5>
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "padding": "24px",
        }
      }
    >
      <div>
        test
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "height": "30%",
          }
        }
      >
        <Blueprint.Icon
          iconName="arrow-right"
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#DDDDDD",
              "borderRadius": "50%",
              "color": "white",
              "cursor": "pointer",
              "display": "flex",
              "height": "58%",
              "justifyContent": "center",
              "margin": "auto",
              "width": "20%",
            }
          }
        />
      </div>
    </div>
    <div
      className="pt-callout"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderTop": "1px solid #DCDCDC",
          "display": "flex",
          "height": "10%",
          "justifyContent": "center",
        }
      }
    />
  </div>
</div>
`;
